src/utils/preprocessRequirement.ts

```diff
@@ -103,7 +112,7 @@ const preprocessRequirement = (requirement: Partial<Requirement>): Requirement =
     processedRequirement.type === "CONTRACT" &&
     Array.isArray(processedRequirement.data?.params)
   ) {
-    processedRequirement.data.params = requirement.data.params.map(
+    processedRequirement.data.params = requirement?.data?.params.map(
```

Comment:
Same here, it's a bit odd that we check if `requirement.data.params` is an array, but we still need optional chaining? 👀